### DISCUSSION TITLE
Supported .NET RabbitMQ.Client.dll For .NET 4.0 Framework

### DISCUSSION THREAD

**Discussion by Pramod34** [rabbitmq/rabbitmq-dotnet-client#1461] [General]
We have a multiple client applications which has lot data transfer among different application. Got we an idea on putting into queue instead everything storing in DB and reading from DB. So we are building a library (.NET dll) with RabbitMQ message broker, so that client applications a can consume dll and send message.

Here is the challenge that I am having, some of our client host/producer application machines have very old machines like windows XP, 7 (having .NET 4.0) and some other are windows servers 2016, 2019, windows 10, 11 (having .NET >= 4.5). As per my research we need to use different RabbitMQ.Client.dll for .NET 4.0 and .NET 4.5 Frameworks. Is there any other way that we can use one .NET C# RabbitMQ.Client.dll for all the environments?

I would really appreciate and encourage any ideas/appoarches/solutions for the above challenge

**Comment by michaelklishin** [comment]
@Pramod34 there likely won't be a single version of this client that would support .NET classic 4.0 and the latest version, for fairly obvious reasons.

This client's [overview doc page](https://www.rabbitmq.com/dotnet.html) has a client/.NET version compatibility table.
